server/lib/middleware.js

```diff
@@ -85,19 +86,36 @@ module.exports = function middlewareConstructor(config) {
       let token = req.user.token = cryptr.decrypt(req.session.token);
 
       if(ASSERT_TOKEN) {
-        // We expect a 64 character HEX string for the token, for example:
-        // '16d4c4d2fa4d6e4aa6b1b5c6a115ad44fd271dd98204f2008bf5efbba5a56dec'
-        let tokenType = typeof token;
-        if(!token) {
-          console.log("ASSERT_TOKEN FAILED: Expected token to exist");
-        } else {
+        /**
+         * Assert whether or not this token is correct or rather broken.
+         * We expect a 64 character HEX string for the token, for example:
+         * '16d4c4d2fa4d6e4aa6b1b5c6a115ad44fd271dd98204f2008bf5efbba5a56dec'
+         */
+        let assert = function(token) {
+          if(!token) {
+            console.log("ASSERT_TOKEN FAILED: Expected token to exist");
+            return false;
+          }
+
+          let tokenType = typeof token;
+          
           if(tokenType !== "string") {
             console.log("ASSERT_TOKEN FAILED: Expected token type to be String, instead got: " + tokenType);
-          } else {
-            if(!/^[a-z0-9]{64}$/.test(token)) {
-              console.log("ASSERT_TOKEN FAILED: Expected token to only have chars a-z, 0-9. Also got: '" +  token.replace(/[a-z0-9]/g, ' ') + "'");
-            }
+            return false;
           }
+          
+          if(!/^[a-z0-9]{64}$/.test(token)) {
+            console.log("ASSERT_TOKEN FAILED: Expected token to only have chars a-z, 0-9. Also got: '" +  token.replace(/[a-z0-9]/g, ' ') + "'");
+            return false;
+          }
+          
+          return true;
+        }
+
+        if (!assert(token)) {
+          console.log("ASSERT_TOKEN FAILED: retrying decryption using eas-256 rather than eas-256-ctr");
+          token = req.user.token = cryptrFallback.decrypt(req.session.token);
+          assert(token);
         }
```

Comment:
At this point, if we still don't have a valid token, I think we should consider calling `next()` with an error that indicates that the user's session has expired, and they need to login again.  This will trigger our `error.html` template to render, and at least give some clue that there is something wrong.